network.py

```diff
@@ -83,8 +127,14 @@ def accuracy(y_true, y_pred):
     return K.mean(K.equal(y_true, K.cast(y_pred < 0.5, y_true.dtype)))
 
 
-def compile(model):
-    opt = RMSprop()
-    # opt = SGD(lr=0.0003, decay=1e-6, momentum=0.9, nesterov=True)
-    # opt = Nadam()
-    model.compile(loss=contrastive_loss, optimizer=opt, metrics=[accuracy])
+def compile(model, optimizer=None, loss_func=False):
```

Comment:
Instead of `False`, we can use `contrastive_loss` as the default directly, so you can also remove the `loss_func = contrastive_loss if loss_func == False else 'binary_crossentropy'` line.